test/python/algorithms/optimizers/test_umda.py

```diff
@@ -0,0 +1,54 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for the UMDA optimizer."""
+
+from test.python.algorithms import QiskitAlgorithmsTestCase
+
+from qiskit.algorithms.optimizers.umda import UMDA
+
+
+class TestUMDA(QiskitAlgorithmsTestCase):
+    """Tests for the UMDA optimizer."""
+
```

Comment:
There is no test that runs the `minimize` to see that everything works as expected given a problem. I mentioned before the other optimizers are tested to find the minimum of a Rosen function. We need something that covers the main function of the optimizer at that level - just doing setters and getters in not enough. And testing, as before using its private methods and internal knowledge is not done in unit tests - all testing is by the the public API. But using minimize, with perhaps various settings, the code should  be able to be tested that it performs as expected (whatever its doing internally to achieve that which from a unit test perspective here we don't care about, just that it all works)

Reply:
In the optimizers test here https://github.com/Qiskit/qiskit-terra/blob/4d251d1c2d683e4bdb29567fa2cc11711c6d3800/test/python/algorithms/optimizers/test_optimizers.py#L90 you can see we test that it finds the point that corresponds to the minimum so we know its working as expected. Can we check here too otherwise all its doing is checking the length of that point, so the optimizer could be updated and it could go from working to stopping early for some reason and we would never notice. The better we can assert that minimize is really working the more value the unit test has in ensuring code works and continues to work.